File: commonlib/src/main/java/jason/jan/commonlib/base/BaseApplication.java
package jason.jan.commonlib.base;

import android.app.Application;

import com.alibaba.android.arouter.BuildConfig;
import com.alibaba.android.arouter.launcher.ARouter;

/**
 * Description: Application基类
 * 加了一个Aouter而已
 * *
 * Creator: Wang
 * Date: 2019/12/18 23:46
 */
public class BaseApplication extends Application {

    private static BaseApplication instance;

    @Override
    public void onCreate() {
        super.onCreate();
        instance=this;
        initRouter();
    }

    @Override
    public void onTerminate() {
        super.onTerminate();
    }

    @Override
    public void onTrimMemory(int level) {
        super.onTrimMemory(level);
    }

    @Override
    public void onLowMemory() {
        super.onLowMemory();
    }

    public static BaseApplication getInstance(){
        return instance;
    }

    private void initRouter(){
        if (BuildConfig.DEBUG) {
            ARouter.openDebug();
            ARouter.openLog();
        }
        ARouter.init(this);
    }

}

File: commonlib/src/main/java/jason/jan/commonlib/base/BaseApplication.kt
package jason.jan.commonlib.base

import android.app.Application

import com.alibaba.android.arouter.BuildConfig
import com.alibaba.android.arouter.launcher.ARouter

/**
 * Description: Application基类
 * 加了一个Aouter而已
 * *
 * Creator: Wang
 * Date: 2019/12/18 23:46
 */
 open class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        instance = this
        initRouter()
    }

    override fun onTerminate() {
        super.onTerminate()
    }

    override fun onTrimMemory(level: Int) {
        super.onTrimMemory(level)
    }

    override fun onLowMemory() {
        super.onLowMemory()
    }

    private fun initRouter() {
        if (BuildConfig.DEBUG) {
            ARouter.openDebug()
            ARouter.openLog()
        }
        ARouter.init(this)
    }

    companion object {

        var instance: BaseApplication? = null
            private set
    }

}
